api4/channel_test.go

```diff
@@ -1481,43 +1481,44 @@ func TestUpdateChannelPrivacy(t *testing.T) {
 
 	for _, tc := range tt {
 		t.Run(tc.name, func(t *testing.T) {
-			_, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
+			_, resp := c.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
 			CheckForbiddenStatus(t, resp)
 		})
 	}
+	th.TestForAllClients(t, func(t *testing.T, client *model.Client4) {
+		th.LoginTeamAdmin()
 
-	th.LoginTeamAdmin()
-
-	tt = testTable{
-		{"Converting default channel to private should fail", defaultChannel, model.CHANNEL_PRIVATE},
-		{"Updating privacy to an invalid setting should fail", publicChannel, "invalid"},
-	}
+		tt := testTable{
+			{"Converting default channel to private should fail", defaultChannel, model.CHANNEL_PRIVATE},
+			{"Updating privacy to an invalid setting should fail", publicChannel, "invalid"},
+		}
 
-	for _, tc := range tt {
-		t.Run(tc.name, func(t *testing.T) {
-			_, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
-			CheckBadRequestStatus(t, resp)
-		})
-	}
+		for _, tc := range tt {
+			t.Run(tc.name, func(t *testing.T) {
+				_, resp := client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
+				CheckBadRequestStatus(t, resp)
+			})
+		}
 
-	tt = testTable{
-		{"Default channel should stay public", defaultChannel, model.CHANNEL_OPEN},
-		{"Public channel should stay public", publicChannel, model.CHANNEL_OPEN},
-		{"Private channel should stay private", privateChannel, model.CHANNEL_PRIVATE},
-		{"Public channel should convert to private", publicChannel, model.CHANNEL_PRIVATE},
-		{"Private channel should convert to public", privateChannel, model.CHANNEL_OPEN},
-	}
+		tt = testTable{
+			{"Default channel should stay public", defaultChannel, model.CHANNEL_OPEN},
+			{"Public channel should stay public", publicChannel, model.CHANNEL_OPEN},
+			{"Private channel should stay private", privateChannel, model.CHANNEL_PRIVATE},
+			{"Public channel should convert to private", publicChannel, model.CHANNEL_PRIVATE},
+			{"Private channel should convert to public", privateChannel, model.CHANNEL_OPEN},
+		}
 
-	for _, tc := range tt {
-		t.Run(tc.name, func(t *testing.T) {
-			updatedChannel, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
-			CheckNoError(t, resp)
-			assert.Equal(t, tc.expectedPrivacy, updatedChannel.Type)
-			updatedChannel, err := th.App.GetChannel(tc.channel.Id)
-			require.Nil(t, err)
-			assert.Equal(t, tc.expectedPrivacy, updatedChannel.Type)
-		})
-	}
+		for _, tc := range tt {
+			t.Run(tc.name, func(t *testing.T) {
+				updatedChannel, resp := c.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```

Comment:
I think the `c` (`th.Client`) here needs to be changed to `client`
```suggestion
				updatedChannel, resp := client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```